pkg/metadataapi/server/api/logs.go

```diff
@@ -0,0 +1,51 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+
+	utilapi "github.com/puppetlabs/horsehead/v2/httputil/api"
+	"github.com/puppetlabs/relay-core/pkg/metadataapi/errors"
+	"github.com/puppetlabs/relay-core/pkg/metadataapi/server/middleware"
+	"github.com/puppetlabs/relay-pls/pkg/plspb"
+)
+
+func (s *Server) PostLogMessage(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	managers := middleware.Managers(r)
+
+	var request plspb.LogMessageAppendRequest
+	if err := json.NewDecoder(r.Body).Decode(&request); err != nil {
```

Comment:
It feels weird to me to accept a protobuf message over a non-protobuf protocol, especially in the case where it's encoding some standard REST-ish headers (and see the other comment about including the log ID in the path...). I think this handler should just accept bytes with a `Content-Type` of `application/octet-stream`, similar to `PutOutput`. What do you think?

Reply:
On the surface I like this suggestion, but it would currently require this to be reflected to the Manager interface and the specific service interface, and that's moving further away from the direction we want to go in. The service interface shouldn't be receiving generic or abstract models.

Reply:
I haven't seen anything that leads me to believe that sending the gRPC models over REST would be an issue. I feel like they are at least attempting to ensure that works properly.